third-culture-software/bhima
# Improved importing of inventories

This PR improves the importing of inventories (inventory definitions, not articles of stock):

- After importing new inventories, it refreshes the inventory registry
- If successful, it reports the number of new inventories that were created
- Improved warnings about bad import files
- Adds the ability to include a column to add a tag for all new inventories

**TESTING**
- Use bhima_test (do 'yarn build:db')
- Go to:  Inventory > Inventory Registry > [Menu] -> Import Inventories
   - Note the recently improved column definitions, including the new optional 'tag' column
   - Try a test file (try the attached test file 'import-inventories-test-good.csv', which should work)
   - Note the success message
   - Try importing the "good" test file again and note that no new inventories are imported
   - There are two "bad" test files.  Please try each.   The importer will be a little more helpful.  It will report if the required column headers are missing or misspelled, it will alert you to that.  If some rows do not have data in the required columns, the importer will complain about that
   - In the Inventory Registry, enable the filter and search the "Tag" column for A to see the tag that was created due to some rows in the "good" import file having values in the "tag" column.
 
[import-inventories-test-good.csv](https://github.com/IMA-WorldHealth/bhima/files/9358095/import-inventories-test-good.csv)
[import-inventories-test-bad-header.csv](https://github.com/IMA-WorldHealth/bhima/files/9358097/import-inventories-test-bad-header.csv)
[import-inventories-test-missing-data.csv](https://github.com/IMA-WorldHealth/bhima/files/9358098/import-inventories-test-missing-data.csv)
.
